### ISSUE TITLE
Increase the number of extruders limit in Cura.

### ISSUE THREAD

**Issue by Nxtmind** [Ultimaker/Cura#12433]
### Is your feature request related to a problem?

Hey Team!

I have recently purchased a Geeetech A10T and have fallen in love with its color mixing functionality. I have successfully configured Cura to support 8 extruders (virtual mixing tools) within the custom printer GUI. I have also attempted to create my own printer definition to support 24 extruders. 

Unfortunately, it seems that 16 extruders is the hard limit for cura to properly slice, adding anymore extruders to the definition causes the slicer to error and respond with `Unable to slice with the current settings. The following settings have errors: Number of Extruders That Are Enabled, Number of Extruders`. I was hoping to get at least 8 shades in between each of the 3 physical extruders (totaling 24 shades).

Digging around in the codebase led me to this commit (confirming my empirical findings). [Parse up to 16 extruders](https://github.com/Ultimaker/Cura/commit/d778b539c29c5890e04601d770c7b664b1a7af0e)

PS. Love the product, been using it since 2014 with my old Printrbot simple metal!

### Describe the solution you'd like

I would love if Cura can natively support an arbitrary amount of extruders. If that is not possible I would appreciate some guidance on how to go about modifying the source and building my own version (unless there are significant technical limitation).

I would really love to be able to assign my models with whatever color I want without going into the machine settings to change the ratios of the mixing. 

### Describe alternatives you've considered

My current workflow is splits the 24 colors into 3 sets of 8 colors 
Red->Yellow, Yellow->Blue, Blue->Red

I then pick the color I want my, and then copy and paste the correct values into my machine settings.
This is unfortunately quite tedious and error prone.

### Affected users and/or printers

This will affect anyone with a color mixing printer!

Printers with Diamond Hotends 

Geeetech A10M
Geeetech A20M
Geeetech A30M

Geeetech A10T
Geeetech A20T
Geeetech A30T

### Additional information & file uploads

I love color mixing! With only 3 spools, I can get a world of colors!
![IMG-2431](https://user-images.githubusercontent.com/3172111/171770558-7c57e4ad-76e1-40d5-9739-d547d232de79.jpg)

**Comment by GregValiant** [comment]
As the multiple-in-one-out extruders get more popular this is something that may need to be addressed.  It isn't going to be as easy as just changing a number to change the limit.  Cura is set up to display a bunch of extruders and to set models to different extruders but there isn't the screen real estate for 24.  Coming up with a new solution (a parking lot for virtual extruders??) will take some thought and time.  So I'll mark this as Deferred for now.  Once things calm down from the release of 5.0 and the problems that arose from the change in dialogs and controls, then maybe the team will be able to take a breath and look at something like this.  One thing to remember though is that all the printers you mentioned are Geeetech printers.  There isn't an Ultimaker among them.

**Comment by nallath** [comment]
Adding support for even more extruders is indeed going to be a _lot_ of work on our side, mostly for the UI/UX. It's just going to be too much work for something that Ultimaker isn't going to use, so this is indeed something that we're going to defer.

**Comment by Nxtmind** [comment]
Thanks for the context!
I ended up having a couple terrible bugs in Cura 5.0 when using 16 extruders. Firstly, the prime tower would not prime any extruders other than the first one. Even worse, selecting an extruder n in the "Prepare" view would result in extruder n-1 being show in the "Preview" view.
After these 2 issues, I have decided to move away from Cura and have experimented with other slicers. 

For anyone that is in the same benchy as me, I have found that using Prusa Slicer with my Geeetech A10T has been an absolute pleasure. I will definitely give Cura another try in the future!
